java/cuvs-java/src/main/java22/com/nvidia/cuvs/internal/CagraIndexImpl.java

```diff
@@ -748,6 +948,25 @@ public static class IndexReference {
     private IndexReference(MemorySegment indexMemorySegment, Dataset dataset) {
       this.memorySegment = indexMemorySegment;
       this.dataset = dataset;
+      this.graphDevicePointer = null;
+      this.graphBytes = 0;
+      this.resources = null;
+    }
+
+    /**
+     * Constructor for graph-based index, keeps graph memory alive.
+     */
+    private IndexReference(
+        MemorySegment indexMemorySegment,
+        Dataset dataset,
+        MemorySegment graphDevicePointer,
+        long graphBytes,
+        CuVSResourcesImpl resources) {
+      this.memorySegment = indexMemorySegment;
+      this.dataset = dataset;
+      this.graphDevicePointer = graphDevicePointer;
```

Comment:
I think this can be avoided if you don't allocate GPU graph memory yourself, keeping things easier/tidier (but better double check)